src/main/java/uk/gov/ons/census/casesvc/service/InvalidAddressService.java

```diff
@@ -0,0 +1,47 @@
+package uk.gov.ons.census.casesvc.service;
+
+import static uk.gov.ons.census.casesvc.utility.JsonHelper.convertObjectToJson;
+
+import java.util.UUID;
+import org.springframework.stereotype.Component;
+import uk.gov.ons.census.casesvc.logging.EventLogger;
+import uk.gov.ons.census.casesvc.model.dto.EventTypeDTO;
+import uk.gov.ons.census.casesvc.model.dto.ResponseManagementEvent;
+import uk.gov.ons.census.casesvc.model.entity.Case;
+import uk.gov.ons.census.casesvc.model.entity.EventType;
+
+@Component
+public class InvalidAddressService {
+  private final CaseService caseService;
+  private final EventLogger eventLogger;
+
+  public InvalidAddressService(CaseService caseService, EventLogger eventLogger) {
+    this.caseService = caseService;
+    this.eventLogger = eventLogger;
+  }
+
+  public void processMessage(ResponseManagementEvent invalidAddressEvent) {
+    // This check shouldn't be needed, but because several different shape messages are getting
+    // published to the same topic according to the event dictionary, let's keep this check in
+    // place so that we know we need to do something about it
+    if (invalidAddressEvent.getEvent().getType() != EventTypeDTO.ADDRESS_NOT_VALID) {
```

Comment:
https://github.com/ONSdigital/census-rm-case-processor/blob/597256fad97835b0eb7a770de249d1323bb400bd/src/main/java/uk/gov/ons/census/casesvc/service/InvalidAddressService.java#L24-L26
Could do with this check across other processors

Reply:
we could put it in, but if we go crazy checking every possible thing then it'll bloat our code. I put this in because we're going to have to handle other messages which arrive on the same topic, but are shaped differently - I imagine that this will get refactored away, so I wanted a comment to warn in the interim